vcd/resource_vcd_nsxt_alb_edgegateway_service_engine_group.go

```diff
@@ -0,0 +1,198 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge Gateway ID in which ALB Service Engine Group should be located",
+			},
+			"service_engine_group_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge Gateway ID in which ALB Service Engine Group should be located",
+			},
+			"max_virtual_services": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "Maximum number of virtual services to be used in this Service Engine Group",
+			},
+			"reserved_virtual_services": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "Number of reserved virtual services for this Service Engine Group",
+			},
+			"deployed_virtual_services": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of deployed virtual services for this Service Engine Group",
+			},
+		},
+	}
+}
+
+func resourceVcdAlbEdgeGatewayServiceEngineGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeAlbServiceEngineGroupAssignmentConfig := getAlbServiceEngineGroupAssignmentType(d)
+	edgeAlbServiceEngineGroupAssignment, err := vcdClient.CreateAlbServiceEngineGroupAssignment(edgeAlbServiceEngineGroupAssignmentConfig)
+	if err != nil {
+		return diag.Errorf("error creating ALB Service Engine Group assignment to Edge Gateway: %s", err)
+	}
+
+	d.SetId(edgeAlbServiceEngineGroupAssignment.NsxtAlbServiceEngineGroupAssignment.ID)
+	return resourceVcdAlbEdgeGatewayServiceEngineGroupRead(ctx, d, meta)
+}
+
+func resourceVcdAlbEdgeGatewayServiceEngineGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeAlbServiceEngineGroupAssignment, err := vcdClient.GetAlbServiceEngineGroupAssignmentById(d.Id())
+	if err != nil {
+		return diag.Errorf("error reading ALB Service Engine Group assignment to Edge Gateway: %s", err)
+	}
+	edgeAlbServiceEngineGroupAssignmentConfig := getAlbServiceEngineGroupAssignmentType(d)
+	// Inject correct ID for update
```

Comment:
```suggestion
	// Add correct ID for update
```
Maybe just `Add` is enough to avoid to much geekness :))))

Reply:
Changed